core/domain/platform_parameter_domain_test.py

```diff
@@ -81,3 +82,1099 @@ def test_to_dict(self):
+
+    def test_evaluate_server_mode_filter(self):
+        filter_dict = {'type': 'server_mode', 'conditions': [('=', 'dev')]}
+        filter_domain = (
+            parameter_domain
+            .PlatformParameterFilter.from_dict(filter_dict))
+
+        dev_context = self.create_example_context(mode='dev')
+        prod_context = self.create_example_context(mode='prod')
+
+        self.assertTrue(filter_domain.evaluate(dev_context))
+        self.assertFalse(filter_domain.evaluate(prod_context))
+
+    def test_evaluate_user_locale_filter(self):
+        filter_dict = {'type': 'user_locale', 'conditions': [('=', 'en')]}
+        filter_domain = (
+            parameter_domain
+            .PlatformParameterFilter.from_dict(filter_dict))
+
+        en_context = self.create_example_context(user_locale='en')
+        zh_context = self.create_example_context(user_locale='zh-hans')
+
+        self.assertTrue(filter_domain.evaluate(en_context))
+        self.assertFalse(filter_domain.evaluate(zh_context))
+
+    def test_evaluate_client_type_filter(self):
+        filter_dict = {'type': 'client_type', 'conditions': [('=', 'Web')]}
+        filter_domain = (
+            parameter_domain
+            .PlatformParameterFilter.from_dict(filter_dict))
+
+        web_context = self.create_example_context(client_type='Web')
+        native_context = self.create_example_context(client_type='Android')
+
+        self.assertTrue(filter_domain.evaluate(web_context))
+        self.assertFalse(filter_domain.evaluate(native_context))
+
+    def test_evaluate_browser_type_filter(self):
+        filter_dict = {'type': 'browser_type', 'conditions': [('=', 'Chrome')]}
+        filter_domain = (
+            parameter_domain
+            .PlatformParameterFilter.from_dict(filter_dict))
+
+        chrome_context = self.create_example_context(browser_type='Chrome')
+        firefox_context = self.create_example_context(browser_type='Firefox')
+
+        self.assertTrue(filter_domain.evaluate(chrome_context))
+        self.assertFalse(filter_domain.evaluate(firefox_context))
+
+    def test_evaluate_app_version_filter_with_eq_comparison(self):
+        filter_dict = {'type': 'app_version', 'conditions': [('=', '1.2.3')]}
+        filter_domain = (
+            parameter_domain
+            .PlatformParameterFilter.from_dict(filter_dict))
+        self.assertTrue(filter_domain.evaluate(
+            self.create_example_context(app_version='1.2.3')))
+        self.assertFalse(filter_domain.evaluate(
+            self.create_example_context(app_version='1.2.4')))
+
+    def test_evaluate_app_version_filter_with_gt_comparison(self):
+        filter_dict = {'type': 'app_version', 'conditions': [('>', '1.2.3')]}
+        filter_domain = (
+            parameter_domain
+            .PlatformParameterFilter.from_dict(filter_dict))
+        self.assertFalse(filter_domain.evaluate(
+            self.create_example_context(app_version='0.2.3')))
```

Comment:
Similarly, please split this and later tests up such that each test is verifying a single situation/behavior.

Reply:
Done, splitted the version related tests to three parts - evaluate with smaller versions, evaluate with same version and evaluate with larger versions.